### PR TITLE
Roll 61 nfc prototype page

### PR DESCRIPTION
**Pull request by brandlxee11**
The card exchange pipeline is in place. Now, it's a matter of actually using NFC for sending the `id` of the person who will receive the card to the user who is requesting to send their card.

### CODE REVIEW

File: src/app/pages/home/exchange/exchange.page.ts [hunk 1]
```diff
@@ -152,11 +169,156 @@ export class ExchangePage implements OnInit {
...
+      let toast = this.toastCtrl.create({
+        message: 'Connected',
+        duration: 1000,
+        position: 'top'
+      }).then(toast => toast.present());
+
+    }, (err) => {
+      console.log('error attaching ndef listener', err);
+
+      let toast = this.toastCtrl.create({
```
Comment (Yogen1337) [inline src/app/pages/home/exchange/exchange.page.ts:215]: Is this needed twice? Lines 206 - 210 seem to be the same.

Comment (Yogen1337) [reply src/app/pages/home/exchange/exchange.page.ts:215]: If so, we could split this into its own function.

File: src/app/pages/home/exchange/exchange.page.ts [hunk 2]
```diff
@@ -152,11 +169,156 @@ export class ExchangePage implements OnInit {
...
+
+
+       if (event.tag.ndefMessage){
+          console.log('received ndefMessage. the tag contains: ', event.tag);
+          let payloadIndexOne = event.tag.ndefMessage[0].payload;
+          let tagContentOwnerId = await this.nfc.bytesToString(payloadIndexOne).substring(3);
+          this.tagOwnerId = tagContentOwnerId;
+
+          let payloadIndexTwo = event.tag.ndefMessage[1].payload;
+          let tagContentCardId = await this.nfc.bytesToString(payloadIndexTwo).substring(3);
```
Comment (Yogen1337) [inline src/app/pages/home/exchange/exchange.page.ts:244]: Is this something we could bundle with line 232? 
ex:
```
getTagEventPayload(event) {
   const id = await this.nfc.bytesToHexString(event.tag.id);
   const contentOwerId = await this.nfc.bytesToString(payloadIndexOne).substring(3);
   const contentCardId = await this.nfc.bytesToString(payloadIndexTwo).substring(3);
   return { id, contentOwnerId, contentCardId }
}
``

Comment (Yogen1337) [reply src/app/pages/home/exchange/exchange.page.ts:244]: That way you can await on this returned object before continuing. Your data should be loaded by then, I think?
```
const tagEventPayload = await getTagEventPayload(event);
if(tagEventPayload) {
 //Share NFC
//Send Card Approval
}

**Review by Yogen1337 — APPROVED** [review]
Good stuff 👍

File: src/app/pages/home/exchange/exchange.page.ts (hunk 1 above)
Comment (brandlxee11) [reply src/app/pages/home/exchange/exchange.page.ts:215]: I think so in order to indicate whether the ndefListener has been properly added or there was an error. I've never gotten the error toast message though. Splitting into a separate function could definitely help with building out the NFC functionality as clean as possible.

File: src/app/pages/home/exchange/exchange.page.ts (hunk 2 above)
Comment (brandlxee11) [reply src/app/pages/home/exchange/exchange.page.ts:244]: Yeah, I see what you're saying. Good thinking! I'll try that this week. I wonder if the data is even available to `return` anything tangible before connecting with the other device. I'll guess we'll see.

**Pull request merged**